src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -277,13 +296,25 @@ private ValueTask SendPrivateAsync(ReadOnlyMemory<byte> buffer, WebSocketMessage
                 return new ValueTask(Task.FromException(exc));
             }
 
-            MessageOpcode opcode =
-                _lastSendWasFragment ? MessageOpcode.Continuation :
-                messageType == WebSocketMessageType.Binary ? MessageOpcode.Binary :
-                MessageOpcode.Text;
+            bool endOfMessage = messageFlags.HasFlag(WebSocketMessageFlags.EndOfMessage);
+            bool disableCompression;
+            MessageOpcode opcode;
 
-            ValueTask t = SendFrameAsync(opcode, endOfMessage, buffer, cancellationToken);
+            if (_lastSendWasFragment)
+            {
+                disableCompression = _lastSendHadDisableCompression;
```

Comment:
If we imagine a situation where a user forgot to state EOM for the previous message, and here in this call the user wants to send a secret without compression so they pass `DisableCompression` -- which will be silently ignored (because the first message didn't contain one). Should we better throw here if the compression state doesn't match? @scalablecory

Reply:
The `messageType` argument has the same problem. Right now I can start sending `WebSocketMessageType.Text` and continue with `WebSocketMessageType.Binary`, but because it's a continuation, it will be ignored. 

I cannot fix it as well without adding another field keeping track of what message type was last sent. Should I fix both cases? Or none? Or just for the compression state?

Reply:
Changing `messageType` behavior would be a breaking change. I would be reluctant to do that, so I'd just only check `DisableCompression` flag. @scalablecory what is your view on that?

Reply:
Agreed. I've added the functionality. If we decide that we should throw for messageType as well, it should be easy enough to add this with a separate PR.